addons/website_event_questions/models/event_event.py

```diff
@@ -32,37 +33,28 @@ def _compute_question_ids(self):
 
         When synchronizing questions:
 
-          * lines that no answer are removed;
+          * lines with no registered answers are removed;
           * type lines are added;
         """
         if self._origin.question_ids:
             # lines to keep: those with already given answers
-            questions_tokeep_ids = self.env['event.registration.answer'].search(
-                [('question_id', 'in', self._origin.question_ids.ids)]
-            ).question_id.ids
+            questions_tokeep = self.env['event.registration.answer'].search([
+                ('question_id', 'in', self._origin.question_ids.ids),
+            ]).question_id
         else:
-            questions_tokeep_ids = []
+            questions_tokeep = None
+
         for event in self:
             if not event.event_type_id and not event.question_ids:
                 event.question_ids = False
                 continue
 
-            if questions_tokeep_ids:
-                questions_toremove = event._origin.question_ids.filtered(lambda question: question.id not in questions_tokeep_ids)
-                command = [(3, question.id) for question in questions_toremove]
+            if questions_tokeep:
+                questions_toremove = event._origin.question_ids - questions_tokeep
+                event.question_ids = [Command.unlink(q.id) for q in questions_toremove]
```

Comment:
Not sure to understand the change, can't we keep the old code ?

Reply:
The use of magic numbers instead of named commands made it so this fix required more time than it should've, because I first had to look around and ultimately ask around what they meant, in order to know what was happening.
Given that I have to diff those lines anyway, figured I'd make them more readable for the next bug fixer.